server/global/users_importation/usersImporter.js

```diff
@@ -0,0 +1,89 @@
+import dateFormat from "dateformat";
+import mysql from "mysql";
+
+import { parseUsersFromCsv } from "./usersParser.js";
+
+export const LOGIN_MAX_LENGTH = 32;
+
+/**
+ * Parse a file and create script to import all users
+ * @param {string} filepath The file to parse
+ * @returns {Promise<string>} The script
+ */
+export function parseAndCreateSqlToInsertAllUsers(filepath) {
+  return new Promise((resolve, reject) => {
+    parseUsersFromCsv(filepath)
+      .then((json) => {
+        resolve(createSqlToInsertAllUsers(JSON.parse(json)));
+      })
+      .catch(reject);
+  });
+}
+
+/**
+ * Create script to import new users
+ * @param {{login,admin}[]} users The imported users list
+ * @returns {string} The sql script
+ */
+export function createSqlToInsertAllUsers(users) {
+  let script = "START TRANSACTION; SET AUTOCOMMIT=0; ";
+
+  script += createSqlToClearRemovedUsers(users);
+
+  script += createSqlToInsertOrUpdateUsers(users);
+
+  script += "COMMIT; SET AUTOCOMMIT=1; ";
+  return script;
+}
+
+/**
+ * Create script to delete all users that have not been reimported
+ * @param {{login, admin}[]} users The imported users list
+ * @returns {string} The sql script
+ */
+function createSqlToClearRemovedUsers(users) {
+  let logins = users
+    .map(({ login }) => mysql.escape(String(login).substr(0, LOGIN_MAX_LENGTH).toLowerCase()))
+    .join(", ");
+
+  if (logins.length === 0) {
+    logins = ["''"];
+  }
+  const date = mysql.escape(dateFormat(new Date(), "yyyy-mm-dd"));
+  return `  DELETE FROM duel \
```

Comment:
Are the `\` necessary with the "`" ?

Reply:
I think no, but it avoid too much spaces